abel/tools/vmi.py

```diff
@@ -243,3 +243,70 @@ def PAD(theta, beta, amplitude):
         amplitude, error_amplitude = np.nan, np.nan
 
     return (beta, error_beta), (amplitude, error_amplitude)
+
+
+def toPES(radial, intensity, wavelength, Vrep, R2E, zoom=1, energy_form='eBE',
+          units='cm'):
+    """
+    Convert speed radial coordinate into electron binding/kinetic energy.
+    Return the photoelectron spectrum (PES).
+
+    NB requires experiment parameters, measurement `wavelength`,
+    `repeller voltage`, radius squared to `energy calibration factor`.
+    These are easily determined by comparing the generated PES
+    with published spectra. e.g. for O :math:`^-` photodetachment, the
+    strongest fine-structure transition occurs at the electron
+    affinity :math:`EA = 11,784.676(7)` cm :math:`^{-1}`. This energy scaling
+    will provide the `R2E` parameter. Values for the ANU experiment
+    are given below.
+
+    Parameters
+    ----------
+    radial : numpy 1D array
+         radial coordinates
+
+    intensity : numpy 1D array
+         intensity values, at the radial array
+
+    wavelength: float
+         measurement wavelength in nm
+	 e.g. `812.51 nm`, for "examples/data/O-ANU1024.txt"
+
+    Vrep: float
+         repeller voltage 
+         e.g. `-98 volts`, for "examples/data/O-ANU1024.txt"
+
+    R2E: float
+         energy calibration factor
+         e.g. `1.148427` for "examples/data/O-ANU1024.txt"
+
+    zoom: float
+         additional factor if experimental image has been zoomed
+
+    energy_form: str
+         ``eBE`` or ``eKE``, electron binding energy or electron kinetic energy
+
+    units: str
+         ``cm-1`` or ``eV``
+          
+    """
+
+    # transform radius to electron binding energy
+    eBE = 1.0e7/wavelength - radial**2*(np.abs(Vrep)/1000)*R2E/100/zoom**2
+
+    if energy_form == 'eKE':
+        eBKE = 1.0e7/wavelength - eBE   # in cm-1
+    else:
+        eBKE = eBE
+
+    if units == 'eV':
+        eBKE /= 8065.541  # convert cm-1 to eV
+
+    # Jacobian correction to intensity, radius has been squared
+    # Not sure why divide, but this is same as for basex?
+    intensity[1:] /= radial[1:]  # 1: to exclude R = 0
```

Comment:
This is interesting. In general the intensity is arbitrary, so I have not given this much thought. I do know that the how the relative spectral peak intensities should behave, and that has been my criterion for the Jacobian.  However, an absolute intensity could be useful.